In [1]:
import numpy as np
from scipy import stats

# Урок 5

## Проверка статистических гипотез. Доверительные интервалы. A/B-тестирование

__Задача 1__

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.


Нулевая гипотеза: шарик диаметром 17мм, альтернативная - диаметр шарика больше 17 мм.
$$T(X) = \dfrac{\overline{X} - a}{\sigma / \sqrt{n}},$$

In [2]:
T = (17.5 - 17)*np.sqrt(100/4)
T

2.5

Нас интересует правостороняя область

In [3]:
alpha = 0.025
t = stats.norm.ppf(1 - alpha)
t

1.959963984540054

гипотезу следует отвергнуть, так как T > t, и значение попадает в критическую область 
$ \left( t_{1 - \alpha}, \infty \right)$

__Задача 2__

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания `a` с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.

$$
P \left(
    \overline{X} - t_{1 - \alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
    \leq a \leq 
    \overline{X} + t_{1 - \alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
\right) = p.
$$


$p = 1 - \alpha$, $t_{1 - \alpha / 2}$ - квантиль нормального распределения, $\sigma = 16$, $\sqrt{n} = \sqrt{256}$

In [4]:
p = 0.95
alpha = 1 - p
t = stats.norm.ppf(1 - alpha/2)
t

1.959963984540054

${\sigma}/{\sqrt{n}}$ - сокращаются, следовательно доверительный интервал равен (80 - t, 80 + t)

__Задача 3__

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
```
202, 203, 199, 197, 195, 201, 200, 204, 194, 190
```
Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?

In [5]:
X = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [6]:
mean = X.mean()
std = X.std(ddof = 1)
n = 10

mean, std

(198.5, 4.453463071962462)

In [7]:
alpha = 0.01
t = stats.t.ppf(1 - alpha/2, df=9)
t

3.2498355440153697

In [8]:
T = (mean - 200)/(std/np.sqrt(10))
T

-1.0651074037450896

Критическая область (-inf, -3.25) и (3.25, +inf), значение T не попадает в этот диапазон, значит утверждение продавца верно